web/src/main/java/org/cbioportal/web/util/StudyViewFilterApplier.java

```diff
@@ -211,22 +219,29 @@ public SampleIdentifier apply(Sample sample) {
             List<String> studyIds = new ArrayList<>();
             List<String> sampleIds = new ArrayList<>();
             studyViewFilterUtil.extractStudyAndSampleIds(sampleIdentifiers, studyIds, sampleIds);
-            List<Integer> ampEntrezGeneIds = copyNumberGeneFilter.getAlterations().stream().filter(a -> 
-                a.getAlteration() == 2).map(CopyNumberGeneFilterElement::getEntrezGeneId).collect(Collectors.toList());
+            
+            List<String> ampGeneIds = copyNumberGeneFilter.getAlterations().stream().filter(a -> 
+            a.getAlteration() == 2).map(CopyNumberGeneFilterElement::getGeneId).collect(Collectors.toList());
+            
+            List<Integer> ampEntrezGeneIds = extractEntrezGeneIds(ampGeneIds);
+            
             List<DiscreteCopyNumberData> ampCNAList = new ArrayList<>();
             if (!ampEntrezGeneIds.isEmpty()) {
                 ampCNAList = discreteCopyNumberService
                     .getDiscreteCopyNumbersInMultipleMolecularProfiles(molecularProfileService.getFirstDiscreteCNAProfileIds(
-                        studyIds, sampleIds), sampleIds, ampEntrezGeneIds, Arrays.asList(2), Projection.ID.name());
+                        studyIds, sampleIds), sampleIds, new ArrayList<>(ampEntrezGeneIds), Arrays.asList(2), Projection.ID.name());
             }
-
-            List<Integer> delEntrezGeneIds = copyNumberGeneFilter.getAlterations().stream().filter(a -> 
-                a.getAlteration() == -2).map(CopyNumberGeneFilterElement::getEntrezGeneId).collect(Collectors.toList());
+            
+            List<String> delGeneIds = copyNumberGeneFilter.getAlterations().stream().filter(a -> 
```

Comment:
The following three lines of code are identical to 222-226 with the exception of alteration value (2, -2), suggest combining into single routine.

Reply:
Updated